src/helper/engine.hpp

```diff
@@ -79,12 +82,18 @@ class engine {
         // NOLINTNEXTLINE(google-runtime-references)
         void get_meta_and_metrics(std::map<std::string_view, std::string> &meta,
             std::map<std::string_view, double> &metrics);
+        auto schema_extraction_sampled()
```

Comment:
I thought we discussed moving the sampler to the service rather than keeping it in the engine, anything preventing you from doing that?

Reply:
Done